### DISCUSSION TITLE
Action not available in status checks

### DISCUSSION THREAD

**Discussion by BryceAshey** [community/community#26668] [Actions]
<p>Hey Folks - I have an action that ran successfully and I’m trying to set it up to validate a PR. When I go to the branch protection and go to setup the status check the action I created does not show there. I did not add specific yml to set a status - is that required?</p>
<p>I read the numerous stackoverflows, <a href="https://docs.github.com/en/github/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks" rel="noopener nofollow ugc">About Status Checks</a>, and <a href="https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-status-checks-before-merging" rel="noopener nofollow ugc">About Protected Branchs</a> and numerous other articles but none of them talk about the yml to stamp a status… Any guidance is welcome.</p>

**Comment by BryceAshey** [comment, marked as answer]
<p>To sortof close this out - the action is firing on pull requests so it’s working and verifying that status passes. I guess specifically setting it in the branch protection is not required once it’s setup in the actions…</p>

**Comment by jsoref** [comment]
<p>The required status checks thing as of late doesn’t automatically show status checks, instead there’s a text area where you can enter a search criteria to get a choice of matching status checks to add.</p>
<p>This isn’t the first question I’ve seen asked about this, and it feels like the docs could use a picture or two.</p>
<p><a href="https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-status-checks-before-merging" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-status-checks-before-merging</a></p>
<p>Here’s a sample of me searching for a status check to require:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/b/8bbc68f2941dd88a37dbeed9c88d10347588abc8.png" data-download-href="/uploads/short-url/jWa29Y8KA6qTDbOe6zBQEv8X8k8.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/2119212/181099121-1920a12e-d345-4e98-b506-2e4d1b7ed628.png" alt="image" data-base62-sha1="jWa29Y8KA6qTDbOe6zBQEv8X8k8" width="689" height="211" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/8/b/8bbc68f2941dd88a37dbeed9c88d10347588abc8_2_689x211.png, https://aws1.discourse-cdn.com/github/optimized/3X/8/b/8bbc68f2941dd88a37dbeed9c88d10347588abc8_2_1033x316.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/8/b/8bbc68f2941dd88a37dbeed9c88d10347588abc8.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/b/8bbc68f2941dd88a37dbeed9c88d10347588abc8_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1240×380 52.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>And once I’ve added it, I can control whether I care about the source of the status check:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/4/344ce91243a6d4a57f9eedb38c78d2e51e3dee68.png" data-download-href="/uploads/short-url/7sFBIlmI5O8CjJmnQRvMGB0sapO.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/2119212/181099123-c926848d-8726-41bf-a622-24b500646969.png" alt="image" data-base62-sha1="7sFBIlmI5O8CjJmnQRvMGB0sapO" width="690" height="318" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/4/344ce91243a6d4a57f9eedb38c78d2e51e3dee68_2_690x318.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/4/344ce91243a6d4a57f9eedb38c78d2e51e3dee68_2_1035x477.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/3/4/344ce91243a6d4a57f9eedb38c78d2e51e3dee68_2_1380x636.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/4/344ce91243a6d4a57f9eedb38c78d2e51e3dee68_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1384×638 90.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>Here, I’m saying that I want a passing :white_check_mark: from <strong>GitHub Actions</strong> for <strong>Check Spelling</strong>.</p>

**Comment by kion-dgl** [comment]
I'm running into an issue where nothing is showing up in the "Status Checks" list rule even when searching. 

![GitHub workflow status check rule](https://github.com/community/community/assets/25621780/e5a26a4c-5dd5-40fc-8b9d-a4dd2e3034e1)

The workflow file is `.github/workflows/ci.yml`, the name is `Astro CI`, it is triggered on `pull_request`, and I have confirmed that it runs on a pull request. But I cannot add a rule, which would require it to pass before merging a PR. 

![GitHub workflow status check rule](https://github.com/community/community/assets/25621780/6314d532-37d9-4683-a532-87b577d06ef9)

**Reply by jsoref** [reply]
I'd file a ticket with https://support.github.com

**Reply by konrad-sol** [reply]
@jsoref  could you share the ticket number?

**Reply by jsoref** [reply]
Sorry for the ambiguity. I wrote `I'd` to mean `If I were you, I would`.

**Reply by konrad-sol** [reply]
My mistake, thanks.
I checked today and it works. Maybe it was connected with [incident](https://www.githubstatus.com/incidents/lg4ndssz2n7l)

**Reply by bantucaravan** [reply]
I am having the exact same issue. To solve it, I had to search by the name of the specific job. I originally was searching for the name of the whole workflow, and that did not work.

**Reply by Dankh917** [reply]
> I am having the exact same issue. To solve it, I had to search by the name of the specific job. I originally was searching for the name of the whole workflow, and that did not work.

seems to be working thanks

**Comment by achobanov** [comment]
I wasn't finding my action because I was searching by the workflow file name. You have to search for the job name as defined in the yaml. Makes sense, but could be more intuitive.

**Reply by Bash360** [reply]
You just saved me hours of confusion, thank you. I agree with you  it is not intuitive.

**Reply by paigemoody** [reply]
Same! Thank you @achobanov !

**Comment by samuelreichor** [comment]
In my case, it only recognized my actions when the action runs on `pull_request`. 

This works:

```yaml
on:
  pull_request:
```

This does NOT work: 

```yaml
on:
  push:
```

**Comment by jawnothin** [comment]
Why doesn't this match against the actual action itself, rather than the individual jobs within it?? The internals of the action I would expect can change over time, where as the name of the action would likely not. I see this as something that could easily break. I could have the same job name on different actions that actually mean completely different things as well...

As others have suggested it's not intuitive at all, as it's pretty ambiguous to what you should put in here without any help text to go with it.

**Comment by onkoe** [comment]
Just wanted to add that this is fucking horrible. 

For future Googlers: "No required checks" -- "No checks have been added". 

<img width="747" height="208" alt="image" src="https://github.com/user-attachments/assets/ed368184-5dae-4b85-b161-4ea77d615b24" />

## Fix

Search for the name of the workflow file. That is, `<FILE>` in `.github/workflows/<FILE>.yml`.

<img width="836" height="251" alt="image" src="https://github.com/user-attachments/assets/076689b7-67c4-4a19-9366-1ad663fb3a72" />

<img width="336" height="210" alt="image" src="https://github.com/user-attachments/assets/ed0e034c-e03e-4392-a529-39f0b52467a4" />

<img width="1554" height="432" alt="image" src="https://github.com/user-attachments/assets/03d53ccf-c07e-4764-b617-62869be06fe3" />

edit: see shit below

**Reply by jsoref** [reply]
You wrote:

> Search for the name of the workflow file. That is, `<FILE>` in `.github/workflows.<FILE>.yml`.

You meant:
> Search for the name of the workflow file. That is, `<FILE>` in `.github/workflows/<FILE>.yml`.

Note that this isn't quite right. If a workflow has a top level `name:` then you should search for the value of that `name:` field, _otherwise_ you'd search for the `<FILE>` part.

**Reply by onkoe** [reply]
Thanks for the correction! diff for anyone confused: 

<img width="842" height="140" alt="image" src="https://github.com/user-attachments/assets/1d290dc6-1574-454d-a68b-00310a1c5461" />

**Reply by jsoref** [reply]
You can/should edit your original message.

**Reply by zaragozamartin91** [reply]
I can confirm ; you need to search for the name of the file
THEN the checks will show up by NAME
<img width="860" height="324" alt="image" src="https://github.com/user-attachments/assets/d74fd536-17fd-4c74-93d6-ceecea899f71" />
This is VERY poor user experience

**Reply by ohadschn** [reply]
The checks that show up by name are still **job** names, right? As far as I can tell, an entire PR workflow cannot be specified.

**Comment by vasilich6107** [comment]
Holy moly
How counterintuitive is the ui

@onkoe thanks for clear explanation
